File: backend/Services/CarService.cs
using System.Collections.Generic;
using System.Linq;
using backend.Dto;
using backend.Exceptions;
using backend.Handlers;
using backend.Interfaces;
using backend.Models;
using Microsoft.AspNetCore.Authorization;
using Microsoft.EntityFrameworkCore;

namespace backend.Services
{
    public class CarService : ICarService
    {
        private readonly IAuthorizationService _authorizationService;
        private readonly CarRepairDbContext _context;
        private readonly IUserContextService _userContextService;

        public CarService(CarRepairDbContext context, IUserContextService userContextService,
            IAuthorizationService authorizationService)
        {
            _context = context;
            _userContextService = userContextService;
            _authorizationService = authorizationService;
        }

        public IEnumerable<Car> GetAllCars()
        {
            var cars = _context.Cars
                .Include(c => c.User)
                .Include(c => c.CarName)
                .Include(c => c.CarProduction)
                .ToList();
            return cars;
        }

        public IEnumerable<Car> GetCarForUser(int id)
        {
            var car = new List<Car>();
            var cars = _context.Cars
                .Include(c => c.CarName)
                .Include(c => c.CarProduction)
                .ToList();
            var count = cars.Count();
            for (var i = 0; i <= count; i++)
            {
                car.Add(cars.Find(x => x.UserId == id));
                cars.Remove(cars.Find(x => x.UserId == id));
            }

            car.RemoveAll(item => item == null);
            if (car is null) throw new NotFound("Nie znaleziono samochodów");

            return car;
        }

        public void UpdateCar(CarUpdate carUpdate, int id)
        {
            var car = _context.Cars.FirstOrDefault(c => c.CarId == id);
            if (car is null) throw new NotFound("Nie znaleziono samochodu");
            var authorizationResult = _authorizationService
                .AuthorizeAsync(_userContextService.User, car, new Operations(CRUD.Update)).Result;

            if (!authorizationResult.Succeeded) throw new Forbid();


            car.CarNameId = carUpdate.CarNameId;
            car.CarProductionId = carUpdate.CarProductionId;
            car.FuelTypeId = carUpdate.FuelTypeId;
            car.CarVin = carUpdate.CarVin.ToUpper();
            car.CarPlates = carUpdate.CarPlates.ToUpper();

            _context.SaveChanges();
        }

        public void NewCar(AddCar addCar)
        {
            var newCar = new Car
            {
                CarNameId = addCar.CarNameId,
                CarProductionId = addCar.CarProductionId,
                FuelTypeId = addCar.FuelTypeId,
                CarVin = addCar.CarVin.ToUpper(),
                CarPlates = addCar.CarPlates.ToUpper(),
                UserId = _userContextService.GetUserId
            };
            _context.Cars.Add(newCar);
            _context.SaveChanges();
        }

        public void NewCarMechanic(CarUpdate addCar)
        {
            var finduser = _context.Users.FirstOrDefault(u => u.UserEmail == addCar.UserEmail);
            var newCar = new Car
            {
                CarNameId = addCar.CarNameId,
                CarProductionId = addCar.CarProductionId,
                FuelTypeId = addCar.FuelTypeId,
                CarVin = addCar.CarVin.ToUpper(),
                CarPlates = addCar.CarPlates.ToUpper(),
                UserId = finduser.UserId
            };
            _context.Cars.Add(newCar);
            _context.SaveChanges();
        }

        public void DeleteCar(int id)
        {
            var car = _context.Cars.FirstOrDefault(u => u.CarId == id);

            if (car is null) throw new NotFound("Nie znaleziono samochodu");

            _context.Cars.Remove(car);
            _context.SaveChanges();
        }
    }
}

File: frontend/vanilla-car/src/Services/CarService.js
import axios from "axios";
import AuthenticateHeader from "./AuthenticateHeader";

// const API_URL = "http://localhost:5000/api/Car";
// const API_URL_2 = "http://localhost:5000/api/CarProduction"
// const API_URL_3 = "http://localhost:5000/api/CarName"
const API_URL = "https://157.245.22.110:8888/api/Car";
const API_URL_2 = "https://157.245.22.110:8888/api/CarProduction"
const API_URL_3 = "https://157.245.22.110:8888/api/CarName"


class CarService {
    getPublicContent() {
        return axios.get(API_URL);
    }

    getProductionYear() {
        return axios.get(API_URL_2, {headers: AuthenticateHeader()});
    }

    getName() {
        return axios.get(API_URL_3 + "/manufacturer", {headers: AuthenticateHeader()});
    }

    getNameModel() {
        return axios.get(API_URL_3, {headers: AuthenticateHeader()});
    }

    getNameModelCar(name) {
        return axios.get(API_URL_3 + "/" + name, {headers: AuthenticateHeader()});
    }

    getCars() {
        return axios.get(API_URL, {headers: AuthenticateHeader()});
    }

    getCar(UserId) {
        return axios.get(API_URL + "/" + UserId, {headers: AuthenticateHeader()});
    }

    deleteCar(CarId) {
        return axios.delete(API_URL + "/" + CarId, {
            headers: AuthenticateHeader(),
        });
    }

    updateCar(
        CarId,
        carNameId,
        carProductionId,
        FuelTypeId,
        CarVin,
        CarPlates
    ) {
        return axios.put(
            API_URL + "/" + CarId,
            {carNameId, carProductionId, FuelTypeId, CarVin, CarPlates},
            {headers: AuthenticateHeader()}
        );
    }

    addCar(carNameId, carProductionId, FuelTypeId, CarVin, CarPlates) {
        return axios.post(
            API_URL,
            {carNameId, carProductionId, FuelTypeId, CarVin, CarPlates},
            {headers: AuthenticateHeader()}
        );
    }

    addCarMechanic(
        carNameId,
        carProductionId,
        FuelTypeId,
        CarVin,
        CarPlates,
        UserEmail
    ) {
        return axios.post(
            API_URL + "/newcarmechanic",
            {

                carNameId,
                carProductionId,
                FuelTypeId,
                CarVin,
                CarPlates,
                UserEmail,
            },
            {headers: AuthenticateHeader()}
        );
    }
}

export default new CarService();
